### PR TITLE
VACMS-7341 Generate top task pages.

### LINKED ISSUE #7341: Create script to generate top task pages for each system.
## Acceptance Criteria
- [ ] When executed  a top task page will be created in draft for each system
- [ ] Standard values should be ____?
- [ ] Should be reusable with other content types

### CMS Team

Please leave only the team that will do this work selected. If you're not sure, it's fine to leave both selected.

- [ ] `Platform CMS Team`
- [x] `Sitewide CMS Team`

### PR DESCRIPTION
**Pull request by swirtSJW**
## Description

closes #7341 

Set to go March 8
runbbook https://app.mural.co/t/vagov6717/m/vagov6717/1643669604223/960c05c3f9f7dbedac43b195cfdd4d84093cd7e1?sender=kevinwalsh8610

## QA steps
As admin:
- [x] Validate 3 hook_updates ran
image
 - [x] Validate [watchdog messages ](https://pr7421-cbt1wtfdbwjnscx18nybjiyeaxokgo2s.ci.cms.va.gov/admin/reports/dblog?type%5B%5D=va_gov_vamc)convey what happened.

image

 - [x] validate 139 [Register for care pages exist](https://pr7421-cbt1wtfdbwjnscx18nybjiyeaxokgo2s.ci.cms.va.gov/admin/content?title=&type=vamc_system_register_for_care&moderation_state=All&owner=All) 
 
image

- [x] validate 139 [Medical records office pages](https://pr7421-cbt1wtfdbwjnscx18nybjiyeaxokgo2s.ci.cms.va.gov/admin/content?title=&type=vamc_system_medical_records_offi&moderation_state=All&owner=All) exist
image

- [x] validate 139 [Billing and insurance pages](https://pr7421-cbt1wtfdbwjnscx18nybjiyeaxokgo2s.ci.cms.va.gov/admin/content?title=&type=vamc_system_billing_insurance&moderation_state=All&owner=All) exist
image

- Visit https://pr7421-cbt1wtfdbwjnscx18nybjiyeaxokgo2s.ci.cms.va.gov/erie-health-care/billing-and-insurance-0 
   - [x] validate attribution is correct.
image
   - [x] validate node title and menu title are correct and match
   - [x] validate field_office, field_administration and menu parent match.
image

**Test Publishing**
- [x] Publish one of the new medical records office nodes and be sure the old VAMC Detail Page is archived and the word REPLACED added to the title
- [x] Publish one of the new billing and insurance nodes and be sure the old VAMC Detail Page is archived and the word REPLACED added to the title
- [x] Publish one of the new register for care nodes and be sure the old VAMC Detail Page is archived and the word REPLACED added to the title

### Definition of Done

- [ ] Documentation has been updated, if applicable.
- [ ] Tests have been added if necessary.
- [ ] Automated tests have passed.
- [ ] Code Quality Tests have passed.
- [ ] Acceptance Criteria in related issue are met.
- [ ] Manual Code Review Approved.

### Select Team for PR review

- [ ] `Platform CMS team`
- [ ] `Sitewide CMS Team`

### Is this PR blocked by another PR?

- [ ] `DO NOT MERGE`

### Does this PR need review from a Product Owner

- [ ] `Needs PO review`

### CMS user-facing annoucement

Is an announcement needed to let editors know of this change?
- [ ] Yes, and it's written in issue ____ and queued for publication.
  - [ ] Merge and ping @ rachel-kauff so she's ready to publish after deployment
- [ ] Yes, but it hasn't yet been written
  - [ ] Don't merge yet -- ping @ rachel-kauff to prompt her to write and queue content
- [ ] No announcement is needed for this code change.
  - [ ] Merge & carry on unburdened by announcements

### CODE REVIEW

**Comment by swirtSJW** [comment]
@kevwalsh This is generating all that do not already exist.  The menu experience is not great.  They get added to the right menu, just not with a specific parent.

**Comment by swirtSJW** [comment]
@kevwalsh Setting the menu placement it kind of ....errr...annoying.  Right now it is setting here (red) one layer to high. 
image
I am assuming it needs to be on the same layer with contact us and policies.  Should it be alpha order there?  
* Billing and insurance
* Contact us
* Medical records office
* Policies
* Register for care

**Comment by JayDarnell** [comment]
@swirtSJW review looks good so far with the one question about call_user_func_array above. QA steps pass as of this moment. If you add additional QA steps let me know and I'll take another look.

**Comment by swirtSJW** [comment]
This  can be approved, but can not be merged until it has been announced and Data Call made.

**Comment by swirtSJW** [comment]
@JayDarnell  Thanks for the review.
You said 

> @swirtSJW review looks good so far with the one question about call_user_func_array above.

But I don't see a question about it.

File: config/sync/user.role.content_publisher.yml [hunk 1]
```diff
@@ -137,14 +137,19 @@ permissions:
...
+  - 'edit any vamc_system_register_for_care content'
   - 'edit any vba_facility content'
   - 'edit any vet_center content'
   - 'edit any vet_center_cap content'
   - 'edit any vet_center_facility_health_servi content'
   - 'edit any vet_center_locations_list content'
   - 'edit any vet_center_mobile_vet_center content'
   - 'edit any vet_center_outstation content'
+  - 'edit any vha_facility_nonclinical_service content'
+  - 'edit terms in health_care_service_taxonomy'
```
Comment (JayDarnell) [inline config/sync/user.role.content_publisher.yml:152]: This is a regression and should be removed.
```suggestion
```

File: tests/phpunit/Security/RolesPermissionsTest.php [hunk 2]
```diff
@@ -569,14 +586,19 @@ public function expectedPerms() {
...
+          'edit any vamc_system_register_for_care content',
           'edit any vba_facility content',
           'edit any vet_center content',
           'edit any vet_center_cap content',
           'edit any vet_center_facility_health_servi content',
           'edit any vet_center_locations_list content',
           'edit any vet_center_mobile_vet_center content',
           'edit any vet_center_outstation content',
+          'edit any vha_facility_nonclinical_service content',
+          'edit terms in health_care_service_taxonomy',
```
Comment (JayDarnell) [inline tests/phpunit/Security/RolesPermissionsTest.php:601]: This is also a regression and needs to be removed
```suggestion
```

**Review by JayDarnell — CHANGES_REQUESTED** [review]
Two regressions to remove

**Comment by JayDarnell** [comment]
Currently failing on two frontend tests @swirtSJW @kevwalsh 

```Given I am at "/static/"                                                                                                   # Drupal\DrupalExtension\Context\MinkContext::assertAtPath()
      Current response status code is 403, but 200 expected. (Behat\Mink\Exception\ExpectationException)```

and 

``` Then the response status code should be 403                                                                                                                     # Drupal\DrupalExtension\Context\MinkContext::assertResponseStatus()
      Current response status code is 200, but 403 expected. (Behat\Mink\Exception\ExpectationException)```

```Failed scenarios:

    features/decoupled.feature:8
    features/decoupled.feature:14```

**Comment by swirtSJW** [comment]
Interesting that /static failed to build.... that is usually a pipeline error.   Lookiing into the phpunit fail right now.

**Comment by swirtSJW** [comment]
phpunit failures  
image

**Comment by JayDarnell** [comment]
@swirtSJW this appears to work perfectly!

File: config/sync/user.role.content_publisher.yml [hunk 3]
```diff
@@ -151,6 +151,7 @@ permissions:
   - 'edit any vet_center_outstation content'
   - 'edit any vha_facility_nonclinical_service content'
   - 'edit any video media'
+  - 'edit terms in health_care_service_taxonomy'
```
Comment (swirtSJW) [inline config/sync/user.role.content_publisher.yml:154]: Bahh. regression. Fixing

**Pull request merged**